Fixed Lunar Battle quest scrolls showing wrong unlock conditions (Issue #10294 )

DoguD opened the pull request:
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #10295

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
Even though the Lunar Battle quest scrolls unlocked according to check-in count, in Shop/Quests, locked Lunar Battle Part 2 & 3 shows a message which implies the scroll will be unlocked when the user finishes the previous quest. This message is removed.
Before: 

![](https://user-images.githubusercontent.com/1495809/39402445-c5d09c2c-4ba2-11e8-8bbd-0bd94f51f4e2.png)

After:
![screen shot 2018-05-03 at 16 51 25](https://user-images.githubusercontent.com/20525079/39580893-ebecc0d6-4ef2-11e8-8d42-fcbb2daed8d4.png)

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 1b2e5d14-e918-44aa-b669-7c309781fdba

Commit:
Fixed Lunar Battle quest scrolls show wrong unlock conditions

In Shop/Quests, the misleading 'To unlock this quest, complete...' popover message removed from Lunar Battle scrolls. They only show check-in condition message.

Commit:
Fixed Lunar Battle quest scrolls show wrong unlock conditions

In Shop/Quests, the misleading 'To unlock this quest, complete...' popover message removed from Lunar Battle scrolls. They only show check-in condition message.

negue commented on website/client/components/shops/quests/index.vue:
```diff
@@ -139,7 +139,7 @@
                     .popover-content-text(v-if='item.locked && item.key === "lostMasterclasser1"') {{ `${$t('questUnlockLostMasterclasser')}` }}
                     .popover-content-text(v-if='item.locked && item.unlockCondition && item.unlockCondition.incentiveThreshold') {{ `${$t('loginIncentiveQuest', {count: item.unlockCondition.incentiveThreshold})}` }}
-                    .popover-content-text(v-if='item.locked && item.previous') {{ `${$t('unlockByQuesting', {title: item.previous})}` }}
+                    .popover-content-text(v-if='item.locked && item.previous && item.key !== "moon2" && item.key !=="moon3"') {{ `${$t('unlockByQuesting', {title: item.previous})}` }}
```
could you move the condition to a method inside of this file, that way if any other quests have to handled like that, the checks are not hidden somewhere in the templates

Commit:
New method added to displaying locked quest popover message in shop

isBuyingDependentOnPrevious () method checks if item.key of quest is in a list of quests whose unlock condition is not dependent on the completition of previous quest.

Commit:
Merge remote-tracking branch 'origin/develop' into develop

DoguD commented:
I have added a new method for doing the check of whether the quest's buying condition is dependent on the completition of a previous quest.
By the way, I am seeing a lot of commits here, did I do something wrong by calling `git fetch upstream`?

negue commented:
@DoguD this happens if you use `develop` as your work-branch but then with `git fetch upstream` you'll have all commits in that branch ^^

could you try to create a branch and push your work to this and then recreate the pullrequest , its the easier way without rebasing (since a lot more could go wrong there)

Commit:
Merge pull request #1 from HabitRPG/develop

test

DoguD commented:
https://github.com/HabitRPG/habitica/pull/10346 is the new pull request.
